src/main/java/com/algorand/algosdk/v2/client/Utils.java

```diff
@@ -1,19 +1,179 @@
+        Set<Long> apps = new HashSet<>();
+        Set<Long> assets = new HashSet<>();
+        Set<String> accts = new HashSet<>();
+
+        for (SignedTransaction txn : txns) {
+            Transaction tx = txn.tx;
+
+            // We're only interested to pull state for app calls
+            if (tx.type != Type.ApplicationCall) {
+                continue;
+            }
+
+            // If this is a create transaction
+            if (tx.applicationId == 0 || tx.applicationId == null) {
+                // Prepare and set param fields for Application being created
+                // from the transaction passed
+                ApplicationParams params = new ApplicationParams();
+                params.creator = tx.sender;
+                params.approvalProgram = tx.approvalProgram.getBytes();
+                params.clearStateProgram = tx.clearStateProgram.getBytes();
+
+                ApplicationStateSchema localState = new ApplicationStateSchema();
+                localState.numByteSlice = tx.localStateSchema.numByteSlice.longValue();
+                localState.numUint = tx.localStateSchema.numUint.longValue();
+                params.localStateSchema = localState;
+
+                ApplicationStateSchema globalState = new ApplicationStateSchema();
+                globalState.numByteSlice = tx.globalStateSchema.numByteSlice.longValue();
+                globalState.numUint = tx.globalStateSchema.numUint.longValue();
+                params.globalStateSchema = globalState;
+
+                Application app = new Application();
+                app.id = Utils.defaultAppId;
+                app.params = params;
+
+                app_infos.add(app);
+            } else {
+                apps.add(tx.applicationId);
+                accts.add(Address.forApplication(tx.applicationId).toString());
+            }
+
+            if (tx.foreignApps.size() > 0) {
+                apps.addAll(tx.foreignApps);
+            }
+
+            if (tx.foreignAssets.size() > 0) {
+                assets.addAll(tx.foreignAssets);
+            }
+
+            if (tx.accounts.size() > 0) {
+                for (Address acct : tx.accounts) {
+                    accts.add(acct.toString());
+                }
+            }
+        }
+
+        for (Long asset : assets) {
+            Response<Asset> ar = client.GetAssetByID(asset).execute();
+            if(ar.isSuccessful()){
+                Asset a = ar.body();
+                accts.add(a.params.creator);
+            }
+        }
+
+        for (Long app : apps) {
+            Response<Application> ar = client.GetApplicationByID(app).execute();
+            if(ar.isSuccessful()){
+                app_infos.add(ar.body());
+                accts.add(ar.body().params.creator.toString());
+            }
+        }
+
+        for (String acct : accts) {
+            Response<Account> ar = client.AccountInformation(new Address(acct)).execute();
+            if(ar.isSuccessful()){
```

Comment:
one thing to make sure: for other SDK implementation, there's a decodeProgram involved here, this is not needed in Java case, right?

Reply:
If I can edit the generator to allow raw bytes when we msgpack encode, this should work without decoding since the java decoder handles it for us.